test/test_integration.py

```diff
@@ -324,3 +327,140 @@ def test_gallery(self):
             '#dropdown .Select-input input'
         ).send_keys(u'北')
         self.snapshot('gallery - chinese character')
+
+    def test_location_link(self):
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            html.Div(id='waitfor'),
+            dcc.Location(id='test-location', refresh=False),
+
+            dcc.Link(
+                html.Button('I am a clickable button'),
+                id='test-link',
+                href='/test/pathname'),
+            dcc.Link(
+                html.Button('I am a clickable hash button'),
+                id='test-link-hash',
+                href='#test'),
+            dcc.Link(
+                html.Button('I am a clickable search button'),
+                id='test-link-search',
+                href='?testQuery=testValue',
+                refresh=False),
+            html.Button('I am a magic button that updates pathname', id='test-button'),
+            html.A('link to click', href='/test/pathname/a', id='test-a'),
+            html.A('link to click', href='#test-hash', id='test-a-hash'),
+            html.A('link to click', href='?queryA=valueA', id='test-a-query'),
+            html.Div(id='test-pathname', children=[]),
+            html.Div(id='test-hash', children=[]),
+            html.Div(id='test-search', children=[]),
+        ])
+
+        @app.callback(
+            output=Output(component_id='test-pathname', component_property='children'),
+            inputs=[Input(component_id='test-location', component_property='pathname')])
+        def update_location_on_page(pathname):
+            assert isinstance(pathname, str)
+
+            return pathname
+
+        @app.callback(
+            output=Output(component_id='test-hash', component_property='children'),
+            inputs=[Input(component_id='test-location', component_property='hash')])
+        def update_location_on_page(hash_val):
+            if hash_val is None:
+                return ''
+
+            return hash_val
+
+        @app.callback(
+            output=Output(component_id='test-search', component_property='children'),
+            inputs=[Input(component_id='test-location', component_property='search')])
+        def update_location_on_page(search):
+            if search is None:
+                return ''
+
+            return search
+
+        @app.callback(
+            output=Output(component_id='test-location', component_property='pathname'),
+            inputs=[Input(component_id='test-button', component_property='n_clicks')],
+            state=[State(component_id='test-location', component_property='pathname')])
+        def update_pathname(n_clicks, current_pathname):
+            if n_clicks is not None:
+                return '/new/pathname'
+
+            return current_pathname
+
+        self.startServer(app=app)
+
+        def _waiter():
```

Comment:
👍  - this is great. can we move this to `test/utils.py` so that we can reuse it elsewhere?